apps/backend/app.py

```diff
@@ -97,21 +108,27 @@ def run_batch(data):
     os.chdir('../..')
 
 ### UTILS
-def run_experiment(experiment_path, config_path):
+def run_experiment(experiment_path, config_path, filetype):
     #make sure that the cwd is ExperimentsFiles/{ExperimentId}
-    # print(f"Current directory {os.getcwd()}")
-    with Popen(["python",experiment_path,config_path], stdout=PIPE, stdin=PIPE, stderr=PIPE,encoding='utf8') as p:
-        try:
-            data = p.communicate()
-            if data[1]:
-                print(f'errors returned from pipe is {data[1]}')
-                return "ERROR"
-        except Exception as e:
-            print(e)
+    if filetype == 'python':
+        with Popen(['python',experiment_path,config_path], stdout=PIPE, stdin=PIPE, stderr=PIPE,encoding='utf8') as p:
+            return get_data(p)
+    elif filetype == 'java':
+        with Popen(['java','-jar',experiment_path,config_path], stdout=PIPE, stdin=PIPE, stderr=PIPE,encoding='utf8') as p:
+            return get_data(p)
+     
+def get_data(p):
+    try:
+        data = p.communicate()
+        if data[1]:
+            print(f'errors returned from pipe is {data[1]}')
             return "ERROR"
-        result = data[0].split('\n')[0]
-        print(f"result data: {result}")
-        return(result)
+    except Exception as e:
+        print(e)
+        return "ERROR"
```

Comment:
Should we return the error to string as well instead of just Error?

Reply:
Good idea for the future, Im pretty sure that it checks for the word ERROR which probably isn't good but should probably go on like a bug fixing branch or something